src/live_restore/live_restore_fs.c

```diff
@@ -589,7 +592,24 @@ __live_restore_fs_fill_holes_on_file_close(WT_FILE_HANDLE *fh, WT_SESSION *wt_se
         __wt_verbose_debug3((WT_SESSION_IMPL *)wt_session, WT_VERB_FILEOPS,
           "Found hole in %s at %ld-%ld during file close. Filling", fh->name, hole->off,
           WT_EXTENT_END(hole));
-        WT_RET(__live_restore_fh_read(fh, wt_session, hole->off, hole->len, buf));
+
+        off_start = hole->off;
+        off_end = hole->off + (wt_off_t)hole->len;
```

Comment:
Thank you Andrew! I have fixed this as suggested. Since`WT_EXTENT_END(hole)` is the offset_end inclusive, where the previous `hole->off + (wt_off_t)hole->len;` is offset_end exclusive, to adjust this change I also modify the while loop from `while (off_start + WT_LIVE_RESTORE_READ_SIZE < off_end)` to `while (off_start + WT_LIVE_RESTORE_READ_SIZE <= off_end)`.